activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -149,6 +149,7 @@ def exec_query(sql, name = "SQL", binds = [], prepare: false)
       # `nil` is the default value and maintains default behavior. If an array of column names is passed -
       # the result will contain values of the specified columns from the inserted row.
       def exec_insert(sql, name = nil, binds = [], pk = nil, sequence_name = nil, returning: nil)
+        validate_returning_support!(returning)
```

Comment:
Two tiny nits here (but since there are already other changes requested I figured I'd chime in 🙈  )
* Should we name the method `validate_returning_support(returning)` instead? I'm not sure how Rails's style guide rules are defined, but I'm used to the rule where `!` methods should only exist if they have a matching "safe" method.
* Can we move the `unless returning` check up to this line, e.g.
```ruby
validate_returning_support!(returning) if returning
```
instead of having it in `#validate_returning_support`? This is totally personal preference on my part, but having to shift focus to a private method requires  mental bandwidth, and if `returning` is nil we don't need to validate anything. As a reader, seeing that we don't need to validate if returning is `nil` directly in `exec_insert` is a little bit easier than having to move down to `#validate_returning_support!` to see what it's doing 😄

Reply:
Yea, I was primarily copying a few existing methods like
https://github.com/rails/rails/blob/d4ca886d5377a8b51e46594148ea466ccf6b6e54/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L295
Though I agree that generally it should be preferable to follow the styleguide

>and if returning is nil we don't need to validate anything.

Actually I'm surprised I didn't do it from the beginning as generally I do tend to design methods which make callers responsible for making sure the method is not called when the input is invalid. Like I would prefer caller doing `post.archieve unless post.archieved?` than `def archieve` to additionally validate the state. I prefer methods to do the action when called without bailing out on some conditions. Sometimes it may lead to a bit of duplication if every caller needs to perform checks on its own but in most cases I still prefer keeping `if` statements out of the methods

We have moved away from the `validate_returning_support` method so it's not applicable here anymore but I appreciate the feedback!